core/src/main/java/discord4j/core/object/entity/Webhook.java

```diff
@@ -64,6 +65,46 @@ public Snowflake getId() {
         return Snowflake.of(data.id());
     }
 
+    public enum Type {
+        /**
+         * Incoming Webhooks can post messages to channels with a generated token
+         */
+        Incoming(1),
+        /**
+         * Channel Follower Webhooks are internal webhooks used with Channel Following
+         * to post new messages into channels
+         */
+        ChannelFollower(2);
+
+        private final int value;
+
+        Type(int value) {this.value = value;}
+
+        public int getValue() {
+            return value;
+        }
+
+        public static Type fromValue(int value) {
+            switch (value) {
+                case 1:
+                    return Type.Incoming;
+                case 2:
+                    return Type.ChannelFollower;
+                default:
+                    throw new IllegalArgumentException("Unexpected webhook type.");
```

Comment:
Can we expect Discord to add another type? In that case I would want to get `UNKNOWN` from getType instead of an IllegalArgumentException-

![image](https://user-images.githubusercontent.com/1908830/88502998-4bb29a80-cf9e-11ea-8dcf-69a450bc87cb.png)